package/kernel/mac80211/patches/subsys/201-mac80211-add-logic-to-skip-useless-VHT-cap-check-in-.patch

```diff
@@ -0,0 +1,37 @@
+From 793bfa410bd16c273beb390818ab44886c680a22 Mon Sep 17 00:00:00 2001
+From: Christian Marangi <ansuelsmth@gmail.com>
+Date: Wed, 14 Jun 2023 10:20:31 +0200
+Subject: [PATCH 2/4] mac80211: add logic to skip useless VHT cap check in
+ ieee80211_sta_cap_rx_bw
+
+Add logic to skip useless VHT cap check in ieee80211_sta_cap_rx_bw in
+the case where HT doesn't indicate support for 40MHz.
+
+If the STA doesn't indicate support for 40MHz then it doesn't make sense
+to check additional support for 80MHz, 80+80MHz or 160MHz
+
+Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
+---
+ net/mac80211/vht.c | 6 ++++++
+ 1 file changed, 6 insertions(+)
+
+diff --git a/net/mac80211/vht.c b/net/mac80211/vht.c
+index 37f4673d4194..b57391a032c8 100644
+--- a/net/mac80211/vht.c
++++ b/net/mac80211/vht.c
+@@ -400,6 +400,12 @@ ieee80211_sta_cap_rx_bw(struct link_sta_info *link_sta)
+ 		goto out;
+ 	}
+ 
++	/* Skip checking VHT cap and set to 20MHz if HT cap doesn't indicate
++	 * support for 40MHz.
++	 */
++	if (!(link_sta->pub->ht_cap.cap & IEEE80211_HT_CAP_SUP_WIDTH_20_40))
++		return IEEE80211_STA_RX_BW_20;
```

Comment:
This causes the following warning:
```
  CC [M]  build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/mac80211-regular/backports-6.11.2/net/mac80211/vht.o
build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/mac80211-regular/backports-6.11.2/net/mac80211/vht.c: In function 'ieee80211_sta_cap_chan_bw':
build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/mac80211-regular/backports-6.11.2/net/mac80211/vht.c:448:24: error: implicit conversion from 'enum ieee80211_sta_rx_bandwidth' to 'enum nl80211_chan_width' [-Werror=enum-conversion]
  448 |                 return IEEE80211_STA_RX_BW_20;
      |                        ^~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

Is the correct return value ` NL80211_CHAN_WIDTH_20`?

Reply:
That's what I changed mine to, but the odd part is that's not even added by the patch (note it occurs twice, and the function name in the patch vs the function name where that warns).